### PR TITLE
Added inverse of incomplete beta Ix(a,b) and Iy(a,b)

### PR DESCRIPTION
**Pull request by Sumegh-git**
This is an initial draft PR for adding the `inverse incomplete beta` from the GW Cran, KJ Martin, GE Thomas paper.

Tests are to be added yet although few trivial tests are already made.
EDIT: Tests added.

### CODE REVIEW

File: src/beta_inc.jl [hunk 1]
```diff
@@ -825,3 +825,125 @@ function beta_inc(a::T, b::T, x::T, y::T) where {T<:Union{Float16, Float32}}
 end
 beta_inc(a::Real, b::Real, x::Real, y::Real) = beta_inc(promote(float(a), float(b), float(x), float(y))...)
 beta_inc(a::T, b::T, x::T, y::T) where {T<:AbstractFloat} = throw(MethodError(beta_inc,(a, b, x, y,"")))
+
+
+#find x ; given p, q, Ix(p,q) = alpha, beta = logabsbeta(p,q)[1]
```
Comment (simonbyrne) [inline src/beta_inc.jl]: keep the symbols consistent with what we had above, i.e.
```math
p = I_x(a,b)
p+q = 1
x+y = 1
```
where `I_x(a,b)` is defined as in https://dlmf.nist.gov/8.17.2

Comment (Sumegh-git) [reply src/beta_inc.jl]: yeah sure! I was going to change that.

**Comment by Sumegh-git** [comment]
cc @simonbyrne

File: src/beta_inc.jl [hunk 2]
```diff
@@ -825,3 +825,119 @@ function beta_inc(a::T, b::T, x::T, y::T) where {T<:Union{Float16, Float32}}
...
+#Volume 26, Number 1, 1977, pages 111-114. 
+
+"""
+    beta_inc_inv(a,b,beta,p)
+
+Computes inverse of incomplete beta function. Given `a`,`b` and ``I_{x}(a,b) = p`` find `x`; also beta = logbeta(a,b)[1]
+"""
+function beta_inc_inv(a::Float64, b::Float64, beta::Float64, p::Float64)
+    sae = -30.0
+    fpu = 10.0^sae
```
Comment (simonbyrne) [inline src/beta_inc.jl]:
```suggestion
    fpu = 1e-30
```

File: src/beta_inc.jl [hunk 3]
```diff
@@ -825,3 +825,119 @@ function beta_inc(a::T, b::T, x::T, y::T) where {T<:Union{Float16, Float32}}
...
+#GW Cran, KJ Martin, GE Thomas, Remark AS R19 and Algorithm AS 109: A Remark on Algorithms AS 63: The Incomplete Beta Integral and AS 64: Inverse of the Incomplete Beta Integeral,
+#Applied Statistics,
+#Volume 26, Number 1, 1977, pages 111-114. 
+
+"""
+    beta_inc_inv(a,b,beta,p)
+
+Computes inverse of incomplete beta function. Given `a`,`b` and ``I_{x}(a,b) = p`` find `x`; also beta = logbeta(a,b)[1]
+"""
+function beta_inc_inv(a::Float64, b::Float64, beta::Float64, p::Float64)
```
Comment (simonbyrne) [inline src/beta_inc.jl]: I think we should change the function signature to:
```julia
(x,y) = beta_inc_inv(a::Float64, b::Float64, p::Float64, q::Float64; lb = logbeta(a,b)[1])
```

**Comment by Sumegh-git** [comment]
I'll change the function signature.. Is there any other comment you want to make?

**Comment by Sumegh-git** [comment]
Done @simonbyrne

File: src/beta_inc.jl [hunk 4]
```diff
@@ -825,3 +825,120 @@ function beta_inc(a::T, b::T, x::T, y::T) where {T<:Union{Float16, Float32}}
...
 beta_inc(a::T, b::T, x::T, y::T) where {T<:AbstractFloat} = throw(MethodError(beta_inc,(a, b, x, y,"")))
+
+#GW Cran, KJ Martin, GE Thomas, Remark AS R19 and Algorithm AS 109: A Remark on Algorithms AS 63: The Incomplete Beta Integral and AS 64: Inverse of the Incomplete Beta Integeral,
+#Applied Statistics,
+#Volume 26, Number 1, 1977, pages 111-114. 
+
+"""
+    beta_inc_inv(a,b,p,q,lb=logbeta(a,b)[1])
+
+Computes inverse of incomplete beta function. Given `a`,`b` and ``I_{x}(a,b) = p`` find `x`.
```
Comment (simonbyrne) [inline src/beta_inc.jl]: I think the function should return a tuple `(x,y)` (to mirror `beta_inc`).

File: src/beta_inc.jl [hunk 5]
```diff
@@ -825,3 +825,120 @@ function beta_inc(a::T, b::T, x::T, y::T) where {T<:Union{Float16, Float32}}
...
+        t = 1.0/(2*qq - 1.0)
+        h = 2.0/(s+t)
+        w = y*sqrt(h+r)/h - (t-s)*(r + 5.0/6.0 - 2.0/(3.0*h))
+        ans = pp/ (pp+qq*exp(w^2))
+    else
+        r = 2.0*qq
+        t = 1.0/(9.0*qq)
+        t = r*(1.0-t+y*sqrt(t))^3
+        if t <= 0.0
+            ans = 1.0 - exp((log((1.0-aa)*qq)+lb)/qq)
```
Comment (simonbyrne) [inline src/beta_inc.jl]:
```suggestion
            ans = -expm1((log((1.0-aa)*qq)+lb)/qq)
```

File: src/beta_inc.jl [hunk 6]
```diff
@@ -825,3 +825,120 @@ function beta_inc(a::T, b::T, x::T, y::T) where {T<:Union{Float16, Float32}}
...
+#Volume 26, Number 1, 1977, pages 111-114. 
+
+"""
+    beta_inc_inv(a,b,p,q,lb=logbeta(a,b)[1])
+
+Computes inverse of incomplete beta function. Given `a`,`b` and ``I_{x}(a,b) = p`` find `x`.
+"""
+function beta_inc_inv(a::Float64, b::Float64, p::Float64, q::Float64; lb = logbeta(a,b)[1])
+    fpu = 1e-30
+    ans = p
```
Comment (simonbyrne) [inline src/beta_inc.jl]: use `x` instead of `ans` to keep consistent.

File: src/beta_inc.jl [hunk 7]
```diff
@@ -825,3 +825,120 @@ function beta_inc(a::T, b::T, x::T, y::T) where {T<:Union{Float16, Float32}}
...
+    if p > 0.5
+        aa = q
+        pp = b
+        qq = a
+        indx = true
+    else
+        aa = p
+        pp = a
+        qq = b
+        indx = false
```
Comment (simonbyrne) [inline src/beta_inc.jl]: lets try to keep all the symbols consistent:
`aa` should be `pp`
`pp` should be `aa`
`qq` should be `bb`

File: src/beta_inc.jl [hunk 8]
```diff
@@ -825,3 +825,120 @@ function beta_inc(a::T, b::T, x::T, y::T) where {T<:Union{Float16, Float32}}
...
+                x = 1.0 - 2.0/(t+1.0)
+            end
+        end
+    end
+
+    #solve x using modified newton-raphson iteration
+
+    r = 1.0 - pp
+    t = 1.0 - qq
+    yprev = 0.0
```
Comment (simonbyrne) [inline src/beta_inc.jl]: similarly, `y`/`yprev` from here down should be named something related to `p`/`pp`

Maybe `pp_approx`/`pp_approx_prev`?

(I don't think the `y` above here is related)

File: src/beta_inc.jl [hunk 9]
```diff
@@ -825,3 +825,120 @@ function beta_inc(a::T, b::T, x::T, y::T) where {T<:Union{Float16, Float32}}
...
+    end
+
+    iex = max(-5.0/pp^2 - 1.0/aa^0.2 - 13.0, -30.0)
+    acu = 10.0^iex
+
+    #iterate
+    while true
+        y = beta_inc(pp,qq,x)[1]
+        xin = x
+        y = (y-aa)*exp(lb+r*log(xin)+t*log(1.0-xin))
```
Comment (simonbyrne) [inline src/beta_inc.jl]:
```suggestion
        y = (y-aa)*exp(lb+r*log(xin)+t*log1p(-xin))
```

File: src/beta_inc.jl (hunk 8 above)
Comment (Sumegh-git) [reply src/beta_inc.jl]: Yes, sure!

**Pull request merged**